compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/CodeGenerator.java

```diff
@@ -79,9 +78,10 @@ public CompiledJarFile generateTestModule(BLangPackage bLangTestablePackage) {
     private CompiledJarFile generate(BPackageSymbol packageSymbol) {
 
         // Split large BIR functions into smaller methods
-        largeMethodOptimizer = new LargeMethodOptimizer(symbolTable);
-        largeMethodOptimizer.splitLargeBIRFunctions(packageSymbol.bir);
+        BIRRecordValueOptimizer recordValueOptimizer = new BIRRecordValueOptimizer();
 
+        // Optimize record value creation for default values - remove unnecessary method call
+        recordValueOptimizer.optimizeNode(packageSymbol.bir);
```

Comment:
This is not specific to JVM right? Then can't we move this to `BIROptimizer`?

Reply:
This is JVM-specific too. This optimization is to reduce unnecessary method calls which will result in the `method too large` issue. But we can move this to `BIROptimizer` as well. But this relies on the temp var optimization. I move this as the last phase of BIR optimization.